File: MSimpleBoard/src/main/java/com/mooop/board/config/DatabaseConfig.java
package com.mooop.board.config;

import javax.sql.DataSource;

import org.apache.commons.dbcp2.BasicDataSource;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class DatabaseConfig {
	
	@Value("${spring.datasource.dbcp2.url}")
	private String url;
	
	@Value("${spring.datasource.dbcp2.driver-class-name}")
	private String driverClassName;
	
	@Value("${spring.datasource.dbcp2.username}")
	private String username;
	
	@Value("${spring.datasource.dbcp2.password}")
	private String password;
	
	
	@Bean(name = "dataSource"  , destroyMethod = "postDeregister")
	public DataSource dataSource() {
		BasicDataSource dataSource = new BasicDataSource();
		dataSource.setUrl(url);
		dataSource.setDriverClassName(driverClassName);
		dataSource.setUsername(username);
		dataSource.setPassword(password);

		/* Connection count */
		dataSource.setInitialSize(10);
		dataSource.setMaxTotal(10);
		dataSource.setMaxIdle(10);
		
		
		dataSource.setTestOnBorrow(true);
		dataSource.setTestOnReturn(false);
		dataSource.setValidationQuery("SELECT 1 FROM DUAL");
		dataSource.setTestWhileIdle(true);
		dataSource.setTimeBetweenEvictionRunsMillis(6*1000);
		dataSource.setPoolPreparedStatements(true);
		dataSource.setMaxOpenPreparedStatements(10);
		dataSource.setRemoveAbandonedOnBorrow(true);
		dataSource.setRemoveAbandonedTimeout(60);
		
		return dataSource;
	}
}

File: kmvc/core/src/main/kotlin/com/mooop/core/config/DatabaseConfig.kt
package com.mooop.core.config

import com.zaxxer.hikari.HikariConfig
import com.zaxxer.hikari.HikariDataSource
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import javax.sql.DataSource

/**
 * Project : kmvc
 * Package :com.mooop.core.config
 * Author :  zinnaworks
 * Date : 28/04/2022
 * Desc :
 */
@Configuration
class DatabaseConfig {

    @ConfigurationProperties(prefix = "spring.datasource.hikari")
    @Bean
    fun hikariConfig():HikariConfig = HikariConfig()

    @Bean
    fun dataSource():DataSource = HikariDataSource(hikariConfig())
}